google/cloud/bigquery/job.py

```diff
@@ -1871,7 +1870,15 @@ def to_api_repr(self):
         ]
 
         configuration = self._configuration.to_api_repr()
-        _helpers._set_sub_prop(configuration, ["copy", "sourceTables"], source_refs)
+        source_table = _helpers._get_sub_prop(
+            self._configuration._properties, ["copy", "sourceTable"]
+        )
+        if source_table:
+            _helpers._set_sub_prop(
```

Comment:
No need for this logic. We can always set the `sourceTables` with only a single item, but if given a job resource that has only `sourceTable` we need to be able to handle that.